homeassistant/components/lgthinq/entity.py

```diff
@@ -0,0 +1,119 @@
+"""Base class for ThinQ entities."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from thinqconnect import ThinQAPIErrorCodes, ThinQAPIException
+from thinqconnect.integration.homeassistant.property import Property as ThinQProperty
+
+from homeassistant.core import callback
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import DeviceDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(kw_only=True)
+class PropertyInfo:
+    """A data class contains an information for creating property."""
+
+    # The property key for use in SDK must be snake_case string.
+    key: str
+
+    # Optional, if the value should be converted before calling api.
+    value_converter: Callable[[Any], Any] | None = None
```

Comment:
Can you elaborate on what kind of data we get from the API? Is it possible to move the conversion to the library?

Reply:
thinqconnect library gives us 'POWER_ON' string. but we need to convert it into true/false for Home Assistant.
Anyway, we are planning to move it into the library in the next modification.

Reply:
Depends on how easy it is to move to the library. If the library can return multiple data types from that function I'd go with that. Otherwise just move the conversion function to the place where you set the state in the entity class, because it's the same every time

Reply:
Understood. Let me move the conversion function to the place where I set the state.

Reply:
If we move this, do we still need this PropertyInfo and extended entity description?